xbmc/cores/VideoPlayer/VideoRenderers/BaseRenderer.cpp

```diff
@@ -267,6 +274,13 @@ RESOLUTION CBaseRenderer::FindClosestResolution(float fps, float multiplier, RES
         current = (RESOLUTION)i;
         curr    = info;
       }
+      // Prefer highest resolution with best matching refreshrate
+      else if ((i_weight == c_weight) && (info.iScreenWidth > curr.iScreenWidth) && (info.iScreenHeight > curr.iScreenHeight))
+      {
+        current = (RESOLUTION)i;
+        curr    = info;
+        CLog::Log(LOGDEBUG, "Current mode is better (higher res): %s", curr.strMode.c_str());
```

Comment:
It would if you have:

1280x720 running
and next it finds 1680x1050
and after that 1920x1080
and then 3840x2160 and so on

I can remove it. Btw. what suddenly comes to my mind: TVs have 3840x2160
and also non native 4096xsomething else ...

2015-09-08 22:31 GMT+02:00 Rainer Hochecker <notifications@github.com>:

> In xbmc/cores/VideoPlayer/VideoRenderers/BaseRenderer.cpp
> <https://github.com/FernetMenta/xbmc/pull/272#discussion_r38976083>:
>
> > @@ -267,6 +274,13 @@ RESOLUTION CBaseRenderer::FindClosestResolution(float fps, float multiplier, RES
> >          current = (RESOLUTION)i;
> >          curr    = info;
> >        }
> > +      // Prefer highest resolution with best matching refreshrate
> > +      else if ((i_weight == c_weight) && (info.iScreenWidth > curr.iScreenWidth) && (info.iScreenHeight > curr.iScreenHeight))
> > +      {
> > +        current = (RESOLUTION)i;
> > +        curr    = info;
> > +        CLog::Log(LOGDEBUG, "Current mode is better (higher res): %s", curr.strMode.c_str());
>
> better than what? doesn't this produce dozens of log entries on every
> switch?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FernetMenta/xbmc/pull/272/files#r38976083>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B